src/test/kotlin/com/israel/teste/services/ContactServiceTest.kt

```diff
@@ -0,0 +1,153 @@
+        val contactsInDatabase = listOf(
+            Contact(1L, "Israel", "israel@gmail.com", "77981460173"),
+            Contact(2L, "José", "zdelivery@gmail.com", "77981460176"))
+        //Aqui estabelecemos o que esperamos de retorno
+        every {contactRepository.findAll()} returns contactsInDatabase
+        //Action
+        //aqui chamamos exatamente o que estamos esperando
+        val  contactsFound = contactService.listAllContacts()
+        //Assert
+        //Aqui validamos que foi encontrado algo, no caso os contatos mockados
+        Assertions.assertNotNull(contactsFound)
+        //Aqui validamos se o numero de contatos encontras foi o esperado
+        Assertions.assertEquals(2,contactsFound.size)
+    }
+
+    @Test
+    fun addContact(){
+        //Arrange
+        val contact = Contact(1L, "Israel", "isantos@gmail.com", "77981460173")
+        every {contactRepository.existsByNameAndEmail(contact.name,contact.email)} returns false
+        every {contactRepository.existsByName(contact.name)} returns false
+        every {contactRepository.existsByEmail(contact.email)} returns false
+        every {contactRepository.save(any())} returns contact
+
+        //Action
+        val savedContact = contactService.addContact(contact)
+
+        //Assert
+        assertEquals(savedContact,contact)
+    }
+
+    @Test
+    fun `Add contacts return Exception Bad Request`(){
+        //Arrange
+        val contact = Contact(1L, "Israel", "isantos@gmail.com", "77981460173")
+        every { contactRepository.existsByNameAndEmail(contact.name, contact.email)} returns true
+        every {contactRepository.existsByName(contact.name)} returns true
+        every {contactRepository.existsByEmail(contact.email)} returns true
+        //Action
+
+        val thrown = assertThrows<ResponseStatusException> { contactService.addContact(contact)}
+        //Assert
+        assertEquals(thrown.message,"${HttpStatus.BAD_REQUEST} \"Existing name or email\"")
+
+    }
+
+
+
+
+
+
+    @Test()
+    fun showContact() {
+        //Arrange
+        val contact = Contact(1, "Israel Santos", "isantos@gmail.com", "77981460173")
+        every {contactRepository.findById(contact.id)}returns (Optional.of(contact))
+        //Action
+        val result = contactService.showContact(contact.id)
+        //Assert
+        Assertions.assertEquals(result,contact)
+    }
+
+    @Test
+    fun deleteContact(){
+        //Arrange
+        val contact = Contact(1L, "Israel Santos", "isantos@gmail.com", "77981460173")
+        every{contactRepository.findById(contact.id)} returns(Optional.of(contact))
+        every{contactRepository.delete(contact)} just runs
+        //Action
+        val thrown = assertThrows<ResponseStatusException> { contactService.deleteContact(contact.id)}
+        //Assert
+        assertEquals(thrown.message,"${HttpStatus.OK} \"The contact was deleted\"")
+    }
+
+
+     //\"The contact was deleted \"
+    @Test
+    fun alterContact(){
```

Comment:
Normalmente é usado o termo update para alterações. Segue da sigla CRUD: Create, Read, **Update**, and Delete